compiler/src/dotty/tools/MainGenericRunner.scala

```diff
@@ -105,17 +105,20 @@ object MainGenericRunner {
     case "-run" :: fqName :: tail =>
       process(tail, settings.withExecuteMode(ExecuteMode.Run).withTargetToRun(fqName))
     case ("-cp" | "-classpath" | "--class-path") :: cp :: tail =>
-      val globdir = cp.replaceAll("[\\/][^\\/]*$", "") // slash/backslash agnostic
-      val (tailargs, cpstr) = if globdir.nonEmpty && classpathSeparator != ";" || cp.contains(classpathSeparator) then
-        (tail, cp)
+      val cpEntries = cp.split(classpathSeparator).toList
+      val singleEntryClasspath: Boolean = cpEntries.nonEmpty && cpEntries.drop(1).isEmpty
```

Comment:
Isn't just `cpEntries.size == 1` enough?

Reply:
That would be okay, although a wildcard classpath directory could have thousands of jar files, so `size` could be expensive.
Should I change it?

Reply:
We can leave it done your way if you feel that it is better.